### ISSUE TITLE
New WebView-based configuration editor

### ISSUE THREAD

**Issue by vector-of-bool** [vector-of-bool/vscode-cmake-tools#513]
Other IDEs and editors have fancy UIs for interacting with CMake. Why don't we?

With the new WebView API, this should be entirely possible.

Things I'd like:

- Cache variable editor
- Kit selector
- Preferences and settings

**Comment by gocarlos** [comment]
is someone working on this? eventually we could sponsor this feature... 
we are mostly interested in configuring the cmake options as qt creator does: 
![image](https://user-images.githubusercontent.com/1331762/69866808-69abec00-12a5-11ea-8c56-b4a373efe60a.png)

currently we are using the settings.json
```json
{
    "cmake.configureSettings": {
        "SYSTEMX_ENABLE_WEBSERVER": true,
        "SYSTEMX_BUILD_SINGLE_APP": true,
        "SYSTEMX_ENABLE_RUNTIME_PYTHON": true,
        "SYSTEMX_ENABLE_DB": true,
    }
}
```

though we only want to change this for the developer and not for all devs working on this project (this file is checked in)

this should be pretty simple -> just parse the build/CMakeCache.txt and create a nice UI in the CmakeTools UI widget

@bobbrow what do you think?

**Comment by bobbrow** [comment]
This is a good idea, but it is not something that we're looking at right now.  The VS Code team has discouraged us from adding UI for settings, so our first inclination is to work with them to see what we can leverage.

For things that don't fit well into that (e.g. cache editing) we can consider adding UI.  We had to do something similar with cpptools for `c_cpp_properties.json`

**Comment by Zingam** [comment]
A similar feature has been added to Visual Studio 2019 in its latest releases.

**Comment by rafmudaf** [comment]
Agreed, a better interactive experience would be helpful and much more useful than listing the CMakeLists for each target in the existing CMake pane. The UI in qt creator listed above or in the official CMake app are perfect examples.

**Comment by gocarlos** [comment]
we started doing this in our fork: 
![image](https://user-images.githubusercontent.com/1331762/72557170-7cec3880-38a0-11ea-957c-51bb95a02e03.png)

**Comment by gocarlos** [comment]
we want to have something like a GUI config page for cmake.

we would like to merge it back if there is interest from upstream, otherwise we will create an cmake-gui extension which uses the apis from this extension... what do you think?

meanwhile you can test it here: https://marketplace.visualstudio.com/items?itemName=gocarlos.cmake-tools

**Comment by bobbrow** [comment]
Yes, there is upstream interest.  I don't think we want to fragment the CMake experience into separate extensions that provide overlapping, but disparate features.

**Comment by mathieu-lemuzic** [comment]
this was merged in the develop branch but not the main branch... could someone have a look at why this awesome feature was scrapped ? https://github.com/microsoft/vscode-cmake-tools/pull/1490/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626
